torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6258,7 +6258,7 @@ def make_bool_mask(shape):
 
         if mask_t.sum() == 0:
             def random_index(shape):
-                return tuple(map(lambda max_idx: random.randint(0, max_idx), shape))
+                return tuple(map(lambda max_idx: random.randrange(0, max_idx), shape))
```

Comment:
Why the change from randint?

Reply:
I think this was a lurking bug! I can't find the CI run anymore that showed this failure, but it looks like this is trying to draw a valid index into a dimension of length max_idx, but randint draws inclusively to its upper bound, so for a length-5 dimension, it sometimes generates 5 as an index and gets an out of bounds issue.

Not totally sure how this wasn't triggering before

Reply:
That is odd but I see what you mean. 

cc @kshitij12345 @pmeier -- we should consider modifying `make_tensor` to return boolean tensors with a mix of true and false inputs, or providing a new `torch.testing.make_boolean_mask` function (name TBD) for this functionality

Reply:
We can construct boolean tensors today but I don't think we guarantee that they have a mix of True and False, right? Like the tensor could be all True or all False?

Reply:
Fair point, but we sample with 50% probability for each

https://github.com/pytorch/pytorch/blob/620dbc43d8e3c836ad8d934987ee2f87fefbad7a/torch/testing/_creation.py#L129-L130

So unless you sample very small tensors, your chance to get an all `False` or `True` tensor back are vanishingly small. Like `0.5 ** (prod(shape) - 1)` small. If one has such a use case, i.e. a small mask that needs to have both `True` and `False`, they are probably better off hard-coding it.

I wouldn't act on this unless we get request for it and even than it is still up for debate IMO.